### PR TITLE
Track week and subject

### PR DESCRIPTION
**Pull request by rpkamp**

### CODE REVIEW

File: tests/AppBundle/LRS/StatementAssemblerTest.php [hunk 1]
```diff
@@ -151,11 +152,13 @@ public function testAssembler()
...
-        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists(IRI::fromString('http://lrs.surfuni.org/context/label')));
-        $this->assertEquals(1, $stmt->getContext()->getExtensions()->offsetGet(IRI::fromString('http://lrs.surfuni.org/context/recipe')));
+        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('course')));
+        $this->assertTrue($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('recipe')));
+        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('referrer')));
+        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('label')));
+        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('week')));
+        $this->assertFalse($stmt->getContext()->getExtensions()->offsetExists($this->getIriFactory()->create('subject')));
+        $this->assertEquals(1, $stmt->getContext()->getExtensions()->offsetGet($this->getIriFactory()->create('recipe')));
```
Comment (seggen-ibuildings) [inline tests/AppBundle/LRS/StatementAssemblerTest.php]: This block of assertions just became even less readable. Maybe you could use StatementSerializer to convert it into JSON and then use `assertJsonStringEqualsJsonString()`...

File: tests/AppBundle/Service/FimFactoryTest.php [hunk 2]
```diff
@@ -24,22 +24,14 @@
...
-        $extensions->attach(IRI::fromString($iris['referrer']), 'http://www.example.com/');
-        $extensions->attach(IRI::fromString($iris['label']), 'Test');
-        $extensions->attach(IRI::fromString($iris['params']), ['id' => 1234]);
+        $extensions->attach($this->getIriFactory()->create('course'), 1);
+        $extensions->attach($this->getIriFactory()->create('recipe'), 2);
+        $extensions->attach($this->getIriFactory()->create('referrer'), 'http://www.example.com/');
+        $extensions->attach($this->getIriFactory()->create('label'), 'Test');
+        $extensions->attach($this->getIriFactory()->create('week'), 'Week 1');
+        $extensions->attach($this->getIriFactory()->create('subject'), 'PHP');
+        $extensions->attach($this->getIriFactory()->create('params'), ['id' => 1234]);
```
Comment (seggen-ibuildings) [inline tests/AppBundle/Service/FimFactoryTest.php]: Here, the StatementSerializer might be useful as well.

File: web/js/tracker.js [hunk 3]
```diff
@@ -2,20 +2,22 @@
     function Tracker() {
         var url = 'https://la-experiment.surfnet.nl/track';
 
-        function track(recipe, verb, object, course, label, params) {
+        function track(recipe, verb, object, course, label, params, week, subject) {
```
Comment (seggen-ibuildings) [inline web/js/tracker.js]: Is this a good idea? I know we have to be backwards compatible, but maybe we can be smart about it... 😬 

How about... the last non-string argument is interpreted as params?

File: app/Resources/views/admin/course/course.html.twig [hunk 4]
```diff
@@ -92,25 +92,26 @@
 
 Algemene tracking pixel (gebruik dit plaatje in een leeromgeving om de pagina te tracken):
 <pre>
-{{ url('track') }}?recipe={{ recipe.id }}&amp;verb=<span class="script-verb">joined</span>&amp;object=<span class="script-activity">course</span>&amp;course={{ course.id }}&amp;label=<span class="script-label">[label]</span><span class="script-id-placeholder">&amp;params[id]=<span class="script-id"></span></span>
+{{ url('track') }}?recipe={{ recipe.id }}&amp;verb=<span class="script-verb">joined</span>&amp;object=<span class="script-activity">course</span>&amp;course={{ course.id }}&amp;label=<span class="script-label">[label]</span>&amp;week=<span class="script-week">[week]</span>&amp;subject=<span class="script-subject">[subject]</span><span class="script-id-placeholder">&amp;params[id]=<span class="script-id"></span></span>
```
Comment (seggen-ibuildings) [inline app/Resources/views/admin/course/course.html.twig]: Please update the generated JavaScript/HTML to match the `la.push()` API again.

Comment (rpkamp) [inline app/Resources/views/admin/course/course.html.twig]: fixed

**Pull request merged**